mdl/src/main/cft/mdlCreateSecurityGroups.yml

```diff
@@ -330,6 +330,108 @@ Resources:
           FromPort: '3306'
           ToPort: '3306'
           CidrIp: 0.0.0.0/0
+  LambdaExecutionRole:
+    Type: 'AWS::IAM::Role'
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - lambda.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      Path: /
+      Policies:
+        - PolicyName: root
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - logs:CreateLogGroup
+                  - logs:CreateLogStream
+                  - logs:PutLogEvents
+                Resource: arn:aws:logs:*:*:*
+              - Effect: Allow
+                Action:
+                  - ec2:DescribeSecurityGroups
+                  - ec2:RevokeSecurityGroupEgress
+                  - ec2:RevokeSecurityGroupIngress
+                  - ec2:DeleteSecurityGroup
+                Resource: '*'
+  DeleteSecurityGroupLambdaFunction:
+    Type: AWS::Lambda::Function
+    Properties:
+      Handler: 'index.handler'
+      Code:
+        ZipFile: |
+          import cfnresponse
+          import boto3
+
+          def handler(event, context):
+              vpc_id = event['ResourceProperties']['VpcId']
+              physical_id = "DeleteEmrDefaultSecurityGroupLambdaPhysicalID"
+              response_data = {}
+              if event['RequestType'] == 'Delete':
+                  emr_sg_default = [
+                      'ElasticMapReduce-Master-Private',
+                      'ElasticMapReduce-ServiceAccess',
+                      'ElasticMapReduce-Slave-Private'
+                  ]
+                  client = boto3.client('ec2')
+                  try:
+                      response = client.describe_security_groups(
+                          Filters=[
+                              {
+                                  'Name': 'vpc-id',
+                                  'Values': [
+                                      vpc_id,
+                                  ]
+                              },
+                              {
+                                  'Name': 'group-name',
+                                  'Values': emr_sg_default
+                              },
+                          ]
+                      )
+                      sgs = response['SecurityGroups']
```

Comment:
Cosmetic: This could be more readable if you use more descriptive names for what those things actually are:

`sgs` could be `security_groups`
`for item in sgs` could be `for security_group` in `security_groups` 
and so on..

Reply:
With that being said, I also understand that in a CFT embedded lambda we are stuck with an upper-limit of 4096 characters so short variable names do make sense.

Reply:
1. rename sgs to security_groups, 
2. keep the item name, use security_group makes the text much more longer and a little bit harder to read as this item used in multiple places